src/routes/files/handlers.js

```diff
@@ -0,0 +1,51 @@
+import { v4 as uuidv4 } from 'uuid';
+import handleErrors from '../../lib/apiErrorHandler';
+import { sequelize, File } from '../../models';
+import s3Uploader from '../../lib/s3Uploader';
+
+const fs = require('fs');
+const fileType = require('file-type');
+const multiparty = require('multiparty');
+
+const namespace = 'SERVICE:FILES';
+
+const logContext = {
+  namespace,
+};
+
+export async function createFileMetaData(originalFileName, s3FileName) {
+  const newFile = {
+    originalFileName,
+    key: s3FileName,
+    FileStatusID: 1,
+  };
+  let file;
+  try {
+    await sequelize.transaction(async (transaction) => {
+      file = await File.create(newFile, transaction);
+    });
+    return file.dataValues;
+  } catch (error) {
+    return error;
+  }
+}
+
+export default async function uploadHandler(req, res) {
+  const form = new multiparty.Form();
+  form.parse(req, async (error, fields, files) => {
+    if (error) {
+      res.status(500).send(error);
+    }
+    try {
+      const { path, originalFilename } = files.File[0];
+      const buffer = fs.readFileSync(path);
+      const type = await fileType.fromFile(path);
+      const fileName = `${uuidv4()}.${type.ext}`;
+      await s3Uploader(buffer, fileName, type);
```

Comment:
I'm a bit worried about the case where we successfully upload to s3 but fail to insert in the database. In that case we'd have a free floating file in s3 not attached to anything.

Reply:
So we could update the DB first, originally set the status to uploading, and then once the upload promise resolves update the status to uploaded or something.

Or, we could upload first and then if the DB update fails delete the object.

I prefer the first approach. Thoughts? Or some other approach?

Reply:
I like the first approach also - setting status to "uploading" and then updating on success or failure.